scripts/lib/mkcloud-libvirt.sh

```diff
@@ -57,36 +57,85 @@ function libvirt_net_start()
         ip link set mtu 9000 dev $dev
     done
 
+    boot_mkcloud=/etc/init.d/boot.mkcloud
+    boot_mkcloud_d="$boot_mkcloud.d"
+    boot_mkcloud_d_cloud="$boot_mkcloud_d/$cloud"
+
     if [ -z "$NOSETUPPORTFORWARDING" ] ; then
+        # FIXME: hardcoded assumptions about admin net host range
         nodehostips=$(seq -s ' ' 81 $((80 + $nodenumber)))
-        cat > /etc/init.d/boot.mkcloud <<EOS
+
+        : ${cloud_port_offset:=1100}
+        mkdir -p $boot_mkcloud_d
+        cat > $boot_mkcloud_d_cloud <<EOS
 #!/bin/bash
+# Auto-generated from $0 on `date`
+
+iptables_unique_rule () {
+    # First argument must be chain
+    if iptables -C "\$@" 2>/dev/null; then
+        echo "iptables rule already exists: \$*"
+    else
+        iptables -I "\$@"
+        echo "iptables -I \$*"
+    fi
+}
+
+# Forward ports to admin server
+for port in 22 80 443 3000 4000 4040; do
+    iptables_unique_rule PREROUTING -t nat -p tcp \\
+        --dport \$(( $cloud_port_offset + \$port )) \\
+        -j DNAT --to-destination $adminip:\$port
+done
 
-iptables -t nat -F PREROUTING
```

Comment:
Or we can just add a command which drops that port.  Either way is better than the sledgehammer of flushing the whole chain, which will break all other clouds on that machine, as well as any other rules which were added outside `mkcloud`.